tests/pass/float.rs

```diff
@@ -148,155 +253,256 @@ fn basic() {
     assert_eq!(34.2f64.abs(), 34.2f64);
 }
 
+/// Test casts from floats to ints and back
+macro_rules! test_ftoi_itof {
+    (
+        f: $fty:ty,
+        i: $ity:ty,
+        // Int min and max as float literals
+        imin_f: $imin_f:literal,
+        imax_f: $imax_f:literal $(,)?
+    ) => {{
+        /// By default we test float to int `as` casting as well as to_int_unchecked
+        fn assert_ftoi(f: $fty, i: $ity, msg: &str) {
+            test_both_cast::<$fty, $ity>(f, i, msg);
+        }
+
+        /// Unrepresentable values only get tested with `as` casting, not unchecked
+        fn assert_ftoi_unrep(f: $fty, i: $ity, msg: &str) {
+            let msg = format!("{} -> {}: {msg}", stringify!($fty), stringify!($ity));
+            assert_eq_msg::<$ity>(f as $ity, i, &msg);
+        }
+
+        /// Int to float checks
+        fn assert_itof(i: $ity, f: $fty, msg: &str) {
+            let itof_msg = format!("{} -> {}: {msg}", stringify!($ity), stringify!($fty));
+            assert_eq_msg::<$fty>(i as $fty, f, &itof_msg);
+        }
+
+        /// Check both float to int and int to float
+        fn assert_bidir(f: $fty, i: $ity, msg: &str) {
+            assert_ftoi(f, i, msg);
+            assert_itof(i, f, msg);
+        }
+
+        /// Check both float to int and int to float
+        fn assert_bidir_unrep(f: $fty, i: $ity, msg: &str) {
+            assert_ftoi_unrep(f, i, msg);
+            assert_itof(i, f, msg);
+        }
+
+        let fbits = <$fty>::BITS;
+        let ibits = <$ity>::BITS;
+        let imax = <$ity>::MAX;
+        let imin = <$ity>::MIN;
+        let izero = <$ity>::ZERO;
```

Comment:
I am still a bit confused by this mix of macros and traits. Since this is a macro, you can use `0 as $ity`, `$ity::MAX` etc, can't you? `BITS` is `8 * size_of::<$ity>()`. `signed` is the only tricky one, but `0.saturating_sub(1) < 0` should work.

Reply:
The trait originally had a few more things because I was hoping to use a generic function rather than a macro, that just unfortunately fell apart because of the need for literals (at least without relying on float to float casting, which is also being tested). I think keeping the trait for the last few things just makes it easier to use in multiple macros (at least for `Float`).

Reply:
Yeah I think keeping the Float trait is fine, but the Int trait IMO isn't carrying its weight.

Reply:
Alright, it's gone